credential.go

```diff
@@ -0,0 +1,84 @@
+package mls
+
+import (
+	"fmt"
+	"github.com/bifurcation/mint/syntax"
+)
+
+type CredentialType uint8
+
+const (
+	credential_basic = iota
+	credential_x509
+)
+
+// enum {
+//       ecdsa_secp256r1_sha256(0x0403),
+//       ed25519(0x0807),
+//       (0xFFFF)
+//   } SignatureScheme;
+type SignatureScheme uint16
+
+// struct {
+//     opaque identity<0..2^16-1>;
+//     SignatureScheme algorithm;
+//     SignaturePublicKey public_key;
+// } BasicCredential;
+type BasicCredential struct {
+	Identity           []byte `tls:"head=2"`
+	SignatureScheme    SignatureScheme
+	SignaturePublicKey []byte `tls:"head=2"`
+}
+
+type X509Credential struct {
+	CertData []byte `tls:"head=3"`
+}
+
+//	struct {
+//		CredentialType credential_type;
+//		select (Credential.credential_type) {
+//			case basic:
+//				BasicCredential;
+//			case x509:
+//				opaque cert_data<1..2^24-1>;
+//		};
+//} Credential;
+type Credential struct {
+	CredentialType  CredentialType
+	BasicCredential BasicCredential
```

Comment:
These should be pointers.  It'll use less memory, and if things go wrong, you'll get NPE instead of accessing bogus data.

Reply:
Nit: I would drop the `Credential` from these members.  So `c.Basic` / `c.X509`.